### PR TITLE
PriceFeedService#applyLatestPrice() refactoring

### PR DESCRIPTION
**Pull request by tau3**
Hello!

I saw a comment that `applyLatestBisqMarketPrice` can take some time and refactored it using parallel streams. So (in theory) it can perform calculation routines faster. State of `PriceFeedService` is updated in old way because the class is not thread-safe.

If the idea of parrallel streams is bad, implemented unit test can be kept anyway and this branch (with implementation of fallback to ordinary streams) can still be considered as a code cleanup attempt.

### CODE REVIEW

**Comment by tau3** [comment]
This build error confuses me: local build with gradle is OK and source analysis tools show nothing bad in mentioned line.

**Comment by ManfredKarrer** [comment]
@tau3 The build errors are related to the -SNAPSHOT resolvement with jitpack. That causes often false positive failures. We will change the project structure and build after the next release so it becomes easier. You can ignore the travis build failure atm. 

Thanks for the PR! I will look a bit later into your PR. 
Will have another big one (DAO code) which will get added in a minute, hope it will not conflict with yours...

File: src/test/java/bisq/core/provider/price/PriceFeedServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                mockTradeStatistics("b", new Date(initialTime), 7),
+                mockTradeStatistics("a", new Date(initialTime), 8));
+
+        List<TradeStatistics2> stats = new ArrayList<>(obsoletes);
+        stats.add(mockTradeStatistics("a", new Date(initialTime + 100), 8));
+        stats.add(mockTradeStatistics("b", new Date(initialTime + 200), 9));
+
+        Collections.shuffle(stats);
+
+        PriceFeedService service = new PriceFeedService(mock(HttpClient.class),
```
Comment (ManfredKarrer) [inline src/test/java/bisq/core/provider/price/PriceFeedServiceTest.java]: Has to be PriceNodeHttpClient instead of HttpClient (was changed in recent PRs)

**Review by ManfredKarrer — COMMENTED** [review]
Hi @tau3 I looked now into the PR...
In Bisq most is single threaded (UI thread). Only several areas like P2p network, persistence or cryptographic operations are multi threaded. The code for setting the Bisq market price from the statistics is running in the UI thread, so the parallel stream and concurrent collections will not have any effect here (at least I am not aware of any). Could you make a performance test to check the old version vs. the new version?

File: src/test/java/bisq/core/provider/price/PriceFeedServiceTest.java (hunk 1 above)
Comment (tau3) [reply src/test/java/bisq/core/provider/price/PriceFeedServiceTest.java]: What dependency should be added?

**Comment by tau3** [comment]
Unfortunately, i can't build bisq-core today due to bisq-p2p errors - some classes (that are actually present) can't be locatet. Is everything OK with bisq-p2p snapshot?

**Comment by ripcurlx** [comment]
@tau3 Just tried out current master and it builds fine. I think your PR build fails as you need to sync it with the current core master branch.

**Pull request closed without merging**